lighthouse-core/gather/computed/trace-of-tab.js

```diff
@@ -106,12 +121,13 @@ class TraceOfTab extends ComputedArtifact {
     // stable-sort events to keep them correctly nested.
     /** @type Array<LH.TraceEvent> */
     const processEvents = trace.traceEvents
+      // @ts-ignore - property chain exists for 'TracingStartedInPage' event.
```

Comment:
hmmm, we shouldn't need the ts-ignore for .pid, what was the error you're suppressing?

Reply:
error TS2532: Object is possibly 'undefined'.

Reply:
Oh because startedInPageEvt is `LH.TraceEvent|undefined` and we're accessing inside a different block :/

mind just casting instead then? `/** @type {LH.TraceEvent} */ (startedInPageEvt).pid` (same for .tid below)

Reply:
Sure, I can annotate. I surprised the compiler hasn't noticed `if (!startedInPageEvt) throw`

Reply:
yeah, they still do a lot simpler CFG work than closure, focusing almost entirely locally, so things like this still trip them up even while being more sophisticated in other ways :)